rust/agents/relayer/src/relayer.rs

```diff
@@ -55,7 +55,15 @@ impl BaseAgent for Relayer {
     where
         Self: Sized,
     {
-        let core = settings.try_into_hyperlane_core(metrics, None).await?;
+        let core = if let Some(ref remotes) = settings.destinationchainnames {
+            let mut v: Vec<&str> = remotes.split(',').collect();
+            v.push(&settings.originchainname);
+            settings
+                .try_into_hyperlane_core(metrics, Some(v.clone()))
+                .await?
+        } else {
+            settings.try_into_hyperlane_core(metrics, None).await?
+        };
```

Comment:
You can just map over destinationchainnames and pass the option. Not super sure on `chain` here, @mattiecnvr can probably double check me.

```suggestion
        let chains: Option<Vec<&str>> = settings.destinationchainnames.map(|remotes| {
            remotes
                .split(",")
                .chain(once(&settings.originchainname))
                .collect()
        });
        let core = settings.try_into_hyperlane_core(metrics, chains).await?;
```

Reply:
This does not compile, going to ship what we have now and can always clean up later